options.go

```
@@ -90,6 +96,8 @@ func WithCustomRetries(retryOnStatus []int, maxRetries uint8) Option {
 
 		if maxRetries == 0 {
 			maxRetries = 1
+		} else if maxRetries > 255 {
```

golangci-lint (1.64.8)

SA4003: no value of type uint8 is greater than math.MaxUint8

(staticcheck)